### PR TITLE
Log traceback, not just error message

### PR DESCRIPTION
**Pull request by tisnik**
## Description

`logging.exception` logs the exception and the traceback, while
`logging.error` only logs the exception. The former is more appropriate
when logging an exception, as the traceback is often useful for debugging.

## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

### CODE REVIEW

File: ols/app/endpoints/ols.py [hunk 1]
```diff
@@ -145,8 +145,8 @@ def generate_response(
                     llm_response["history_truncated"],
                 )
             except Exception as summarizer_error:
-                logger.error("Error while obtaining answer for user question")
-                logger.error(summarizer_error)
+                logger.exception("Error while obtaining answer for user question")
+                logger.exception(summarizer_error)
```
Comment (bparees) [inline ols/app/endpoints/ols.py:149]: i agree we want the traceback, but do we really want it twice?

File: ols/app/endpoints/ols.py [hunk 2]
```diff
@@ -187,8 +187,8 @@ def validate_question(conversation_id: str, llm_request: LLMRequest) -> str:
         )
     except Exception as validation_error:
         metrics.llm_calls_failures_total.inc()
-        logger.error("Error while validating question")
-        logger.error(validation_error)
+        logger.exception("Error while validating question")
+        logger.exception(validation_error)
```
Comment (bparees) [inline ols/app/endpoints/ols.py:191]: same issue here with having it twice

**Review by bparees — COMMENTED** [review]
can you give us an example of what the logs look like w/ these changes?

File: ols/app/endpoints/ols.py (hunk 1 above)
Comment (tisnik) [reply ols/app/endpoints/ols.py:149]: aaah, good catch!. Yes the 1st might be just `error`

**Comment by tisnik** [comment]
> can you give us an example of what the logs look like w/ these changes?

```
2024-02-20 18:46:39,608 [ols.app.endpoints.ols:ols.py:216] ERROR: Error while validating question
2024-02-20 18:46:39,608 [ols.app.endpoints.ols:ols.py:217] ERROR: Network error has occurred (target https://bam-api.res.ibm.com/v2/text/generation/limits?version=2023-11-22).
Traceback (most recent call last):
  File "/home/ptisnovs/lightspeed-service/ols/app/endpoints/ols.py", line 207, in validate_question
    return question_validator.validate_question(conversation_id, llm_request.query)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ptisnovs/lightspeed-service/ols/src/query_helpers/question_validator.py", line 71, in validate_question
    response = llm_chain(inputs={"query": query})
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ptisnovs/.local/lib/python3.11/site-packages/langchain_core/_api/deprecation.py", line 145, in warning_emitting_wrapper
    return wrapped(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ptisnovs/.local/lib/python3.11/site-packages/langchain/chains/base.py", line 363, in __call__
    return self.invoke(
           ^^^^^^^^^^^^
  File "/home/ptisnovs/.local/lib/python3.11/site-packages/langchain/chains/base.py", line 162, in invoke
    raise e
  File "/home/ptisnovs/.local/lib/python3.11/site-packages/langchain/chains/base.py", line 156, in invoke
    self._call(inputs, run_manager=run_manager)
  File "/home/ptisnovs/.local/lib/python3.11/site-packages/langchain/chains/llm.py", line 103, in _call
    response = self.generate([inputs], run_manager=run_manager)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ptisnovs/.local/lib/python3.11/site-packages/langchain/chains/llm.py", line 115, in generate
    return self.llm.generate_prompt(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ptisnovs/.local/lib/python3.11/site-packages/langchain_core/language_models/llms.py", line 568, in generate_prompt
    return self.generate(prompt_strings, stop=stop, callbacks=callbacks, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
...
...
```

File: ols/app/endpoints/ols.py (hunk 1 above)
Comment (tisnik) [reply ols/app/endpoints/ols.py:149]: updated @bparees

File: ols/app/endpoints/ols.py (hunk 2 above)
Comment (tisnik) [reply ols/app/endpoints/ols.py:191]: also updated

**Pull request merged**